### PR TITLE
Create/Search/Update Employer Users

### PR DESCRIPTION
**Pull request by devJUKI**

### CODE REVIEW

File: src/CSI.IBTA.Administrator/Clients/UserServiceClient.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                _logger.LogError("UserServiceApiUrl is missing in appsettings.json");
+                throw new InvalidOperationException("AuthServiceApiUrl is missing in appsettings.json");
+            }
+            _httpClient.BaseAddress = new Uri(userServiceApiUrl);
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        public async Task<List<Employer>?> GetEmployers(string token)
+        {
+            if (_httpContextAccessor.HttpContext == null)
```
Comment (radislavkireev) [inline src/CSI.IBTA.Administrator/Clients/UserServiceClient.cs]: This comment is related to the classes EmployerClient, EmployerUserClient and UserServiceClient, I just do not want to repeat it 3 times. 

 _httpContextAccessor is not using and could be removed. 

Token receiving logic is repeatable. Lets consider reusing it with inheritance or any other approach you design.

File: src/CSI.IBTA.Administrator/Controllers/HomeController.cs [hunk 2]
```diff
@@ -28,7 +36,22 @@ public IActionResult Index()
                 return RedirectToAction("Logout", "Auth");
             }
 
-            return View();
+            var Employers = await _userServiceClient.GetEmployers(token);
```
Comment (radislavkireev) [inline src/CSI.IBTA.Administrator/Controllers/HomeController.cs]:
```suggestion
            var employers = await _userServiceClient.GetEmployers(token);
```

**Review by radislavkireev — APPROVED** [review]

File: src/CSI.IBTA.UserService/Services/UsersService.cs [hunk 3]
```diff
@@ -161,24 +163,85 @@ public async Task<GenericResponse<UpdatedUserDto>> UpdateUser(int userId, Update
 
             _unitOfWork.Users.Upsert(user);
             await _unitOfWork.CompleteAsync();
-            return new GenericResponse<UpdatedUserDto>(false, null, _mapper.Map<UpdatedUserDto>(user));
+            return new GenericHttpResponse<UpdatedUserDto>(false, null, _mapper.Map<UpdatedUserDto>(user));
+        }
+
+        public async Task<GenericHttpResponse<UpdatedUserDto>> PutUser(int userId, PutUserDto putUserDto)
```
Comment (GytisMas) [inline src/CSI.IBTA.UserService/Services/UsersService.cs:130]: Possibly redundant to have both PutUser and UpdateUser

File: src/CSI.IBTA.UserService/Services/EmployersService.cs [hunk 4]
```diff
@@ -108,50 +109,63 @@ public async Task<GenericResponse<EmployerDto>> UpdateEmployer(int employerId, U
...
+            return new GenericHttpResponse<bool>(false, null, true);
         }
 
-        public async Task<GenericResponse<EmployerDto>> GetEmployer(int employerId)
+        public async Task<GenericHttpResponse<EmployerDto>> GetEmployer(int employerId)
         {
             var e = await _unitOfWork.Employers.GetById(employerId);
 
-            if (e == null) return new GenericResponse<EmployerDto>(true, new HttpError("Emplyer not found", HttpStatusCode.NotFound), null);
+            if (e == null) return new GenericHttpResponse<EmployerDto>(true, new HttpError("Emplyer not found", HttpStatusCode.NotFound), null);
```
Comment (GytisMas) [inline src/CSI.IBTA.UserService/Services/EmployersService.cs]: Typo in error message

File: src/CSI.IBTA.UserService/Services/EmployersService.cs [hunk 5]
```diff
@@ -108,50 +109,63 @@ public async Task<GenericResponse<EmployerDto>> UpdateEmployer(int employerId, U
...
+            return new GenericHttpResponse<EmployerDto>(false, null, _mapper.Map<EmployerDto>(e));
         }
 
-        public async Task<GenericResponse<bool>> DeleteEmployer(int employerId)
+        public async Task<GenericHttpResponse<bool>> DeleteEmployer(int employerId)
         {
             var e = await _unitOfWork.Employers.GetById(employerId);
 
-            if (e == null) return new GenericResponse<bool>(true, new HttpError("Emplyer not found", HttpStatusCode.NotFound), false);
+            if (e == null) return new GenericHttpResponse<bool>(true, new HttpError("Emplyer not found", HttpStatusCode.NotFound), false);
```
Comment (GytisMas) [inline src/CSI.IBTA.UserService/Services/EmployersService.cs]: Typo in error message

File: src/CSI.IBTA.Shared/DTOs/UserDto.cs [hunk 6]
```diff
@@ -2,9 +2,10 @@
...
     public record UpdatedUserDto(int Id, string UserName, string Password, string FirstName, string LastName, int AccountId, Role Role, string PhoneNumber, string EmailAddress, string AddressState, string AddressStreet, string AddressCity, string AddressZip);
     public record CreateUserDto(string UserName, string Password, string FirstName, string LastName, Role Role, int? EmployerId, string PhoneNumber, string EmailAddress, string AddressState, string AddressStreet, string AddressCity, string AddressZip);
     public record UpdateUserDto(string? UserName, string Password, string FirstName, string LastName, string PhoneNumber, string EmailAddress, string AddressState, string AddressStreet, string AddressCity, string AddressZip);
-}
+    public record PutUserDto(string UserName, string Password, string FirstName, string LastName, string PhoneNumber, string EmailAddress, string AddressState, string AddressStreet, string AddressCity, string AddressZip);
```
Comment (GytisMas) [inline src/CSI.IBTA.Shared/DTOs/UserDto.cs:9]: Possibly redundant to have both PutUserDto and UpdateUserDto

**Review by GytisMas — CHANGES_REQUESTED** [review]

**Review by GytisMas — APPROVED** [review]

**Pull request merged**